examples/httpserver_form_data.py

```diff
@@ -0,0 +1,66 @@
+# SPDX-FileCopyrightText: 2022 Dan Halbert for Adafruit Industries
+#
+# SPDX-License-Identifier: Unlicense
+
+import socketpool
+import wifi
+
+from adafruit_httpserver import Server, Request, Response, GET, POST
+
+
+pool = socketpool.SocketPool(wifi.radio)
+server = Server(pool, debug=True)
+
+
+FORM_HTML_TEMPLATE = """
+<html lang="en">
+    <head>
+        <title>Form with {enctype} enctype</title>
+    </head>
+    <body>
+        <a href="/form?enctype=application/x-www-form-urlencoded">
+            <button>Load <strong>application/x-www-form-urlencoded</strong> form</button>
+        </a><br />
+        <a href="/form?enctype=multipart/form-data">
+            <button>Load <strong>multipart/form-data</strong> form</button>
+        </a><br />
+        <a href="/form?enctype=text/plain">
+            <button>Load <strong>text/plain</strong> form</button>
+        </a><br />
+
+        <h2>Form with {enctype} enctype</h2>
+        <form action="/form" method="post" enctype="{enctype}">
+            <input type="text" name="something" placeholder="Type something...">
+            <input type="submit" value="Submit">
+        </form>
+        {submitted_value}
+    </body>
+</html>
+"""
+
+
+@server.route("/form", [GET, POST])
+def form(request: Request):
+    """
+    Serve a form with the given enctype, and display back the submitted value.
+    """
+    enctype = request.query_params.get("enctype", "text/plain")
+
+    if request.method == POST:
+        posted_value = request.data.get("something")
```

Comment:
I got an exception here for request not having property `data` I think maybe this is supposed to be `form_data` instead?

Reply:
You are right, this should be `form_data`. Already fixed.